File: C#4/Triangle.cs
namespace FiguresLib
{
    public class Triangle : Figure //наслідування
    {
      
        public Triangle() : base() { } //коли визив коннструктор визив реалізацію конструктора базового класу

        public Triangle(double ax1, double ay1, double ax2, double ay2, double ax3, double ay3)
            : base(ax1, ay1, ax2, ay2, ax3, ay3) { }

        public Triangle(Triangle other) : base(other) { }

        public double Perimeter()
        {
            return SideLength(x1, y1, x2, y2) +
                   SideLength(x2, y2, x3, y3) +
                   SideLength(x3, y3, x1, y1);
        }

        public double Area()
        {
            double a = SideLength(x1, y1, x2, y2);
            double b = SideLength(x2, y2, x3, y3);
            double c = SideLength(x3, y3, x1, y1);
            double s = (a + b + c) / 2;
            return Math.Sqrt(s * (s - a) * (s - b) * (s - c));
        }
    }
}

File: C++4/Triangle.cpp
#include "Triangle.h"
#include <cmath>

// Конструктор за замовчуванням
Triangle::Triangle() : Figure() {}

// Конструктор з параметрами (явний виклик конструктора базового класу)
Triangle::Triangle(double ax1, double ay1, double ax2, double ay2, double ax3, double ay3)
    : Figure(ax1, ay1, ax2, ay2, ax3, ay3) {}

// Конструктор копіювання
Triangle::Triangle(const Triangle& other) : Figure(other) {}

// Деструктор
Triangle::~Triangle() {}

// Обчислення периметра
double Triangle::perimeter() const {
    return sideLength(x1, y1, x2, y2) +
        sideLength(x2, y2, x3, y3) +
        sideLength(x3, y3, x1, y1);
}

// Обчислення площі
double Triangle::area() const {
    double a = sideLength(x1, y1, x2, y2);
    double b = sideLength(x2, y2, x3, y3);
    double c = sideLength(x3, y3, x1, y1);
    double s = (a + b + c) / 2;
    return sqrt(s * (s - a) * (s - b) * (s - c));
}
